env0/resource_environment.go

```diff
@@ -433,7 +433,7 @@ func setEnvironmentSchema(ctx context.Context, d *schema.ResourceData, environme
 
 	setEnvironmentConfigurationSchema(ctx, d, configurationVariables)
 
-	if d.Get("variable_sets") != nil {
+	if variableSets, ok := d.GetOk("variable_sets"); variableSets != nil && ok {
```

Comment:
same as "configuration" - if this field isn't used - ignore any use-cases related to it.
this will avoid possible conflicts with explicit assignment resource.

Reply:
Sorry if it's a basic question but I'm pretty new to go and terraform provider so I would like to make sure 😅 Let's say we have an environment and we assign a variable set to it (by FE or API) will we see the drift there?

Reply:
I see, I talked about it with @GiliFaroEnv0 and it looked weird that we won't show DD but if that's how we do it for configuration it might be the way 🤔 
I will raise it with the team and update you 😄

Reply:
we are not hiding drifts...
you can either use the assignment resource or set the sets inline in the environment.
You cannot use both. Otherwise, you will have constant drifts/conflicts. (Same with configuration).

It just tests which one is used... if a resource assignments are used, it ignores anything inline in the environment.

Reply:
maybe we just don't understand the code...
how are you checking if another resource is used?

Reply:
It's simple (but not perfect):

if "variables-set" is set in the state, it means we use the "variables-set" inline use-case.
Otherwise, it's ignored.